### ISSUE TITLE
NER Hypernymy Model

### ISSUE THREAD

**Issue by sahmed53gsu** [sahmed53gsu/UMLS#1]
Working on NER Hypernymy model

# Order of tasks:
PRN = medical lingo for as needed

FIRST READ THE README in the ~/project/build_NER_hypernymy folder in the SaturnCloud
See next comment for login details

1. Larry - Run SP trainer on full UMLS dataset from our AWS s3 account, this will yield all sentences with Hypernym pairs.

2. Saadh, me, Larry, Talar? Label the dataset, this can be done automatically, or via prodigy. I would recommend prodigy unless you understand how to automate labeling the data for spaCy. If with prodigy, I would recommend all of us doing that, we only need to annotate enough to get to an accuracy. So we can run a training epoch, see accuracy (F-score) and add more from gold standard dataset PRN.

3. Saadh - train model with spaCy, fine-tune hyperparameters PRN for accuracy.

4. Measure accuracy.

# Resources:
Saadh - please feel free to use the SaturnCloud cluster with whatever resources you need for this. I am happy to pay for that and prefer you to do this. Login: swanson.eric.karl@gmail.com, pass: Z3902lkmj!

**_See the folder in the SaturnCloud titled: build_NER_hypernyms_**

**1. spaCy (use v3 +)**
   Download
       $pip install -U pip setuptools wheel
       $pip install -U spacy[cuda102,transformers,lookups] 
       $python -m spacy download en_core_web_trf
*for spaCy download ensure that the cuda version is the correct version with command line "nvidia-smi" check versioning, can also use conda install from website*
https://spacy.io/usage
https://spacy.io/usage
https://spacy.io/usage/training
https://spacy.io/usage/embeddings-transformers

**2. Prodigy**
https://prodi.gy/
https://prodi.gy/docs
https://support.prodi.gy/

The prodigy wheel for linux, max and windows are in the ~/project/prodigy folder in this SaturnCloud 
The license to use this for annotators is also in that folder

**Comment by kswanjitsu** [comment]
PRN = medical lingo for as needed

1. Me or Larry - Run Larry's SP trainer on full UMLS dataset from our AWS s3 account, this will yield all sentences with Hypernym pairs.

2. Saadh, me, Larry, Talar? Label the dataset, this can be done automatically, or via prodigy. I would recommend prodigy unless you understand how to automate labeling the data for spaCy. If with prodigy, I would recommend all of us doing that, we only need to annotate enough to get to an accuracy. So we can run a training epoch, see accuracy (F-score) and add more from gold standard dataset PRN.

3. Saadh - train model with spaCy, fine-tune hyperparameters PRN for accuracy.

4. Measure accuracy.

**Comment by kswanjitsu** [comment]
# Resources:
Saadh - please feel free to use the SaturnCloud cluster with whatever resources you need for this. I am happy to pay for that and prefer you to do this. Login: swanson.eric.karl@gmail.com, pass: Z3902lkmj!

**_See the folder in the SaturnCloud titled: build_NER_hypernyms_**

**1. spaCy (use v3 +)**
   Download
       $pip install -U pip setuptools wheel
       $pip install -U spacy[cuda102,transformers,lookups] 
       $python -m spacy download en_core_web_trf
*for spaCy download ensure that the cuda version is the correct version with command line "nvidia-smi" check versioning, can also use conda install from website*
https://spacy.io/usage
https://spacy.io/usage
https://spacy.io/usage/training
https://spacy.io/usage/embeddings-transformers

**2. Prodigy**
https://prodi.gy/
https://prodi.gy/docs
https://support.prodi.gy/

The prodigy wheel for linux, max and windows are in the ~/project/prodigy folder in this SaturnCloud 
The license to use this for annotators is also in that folder
